code/modules/cm_aliens/structures/special/egg_morpher.dm

```diff
@@ -183,6 +183,9 @@
 	..()
 
 /obj/effect/alien/resin/special/eggmorph/attack_ghost(mob/dead/observer/user)
+	if(alert(user, "Are you sure you want to become a facehugger?", "Confirmation", "Yes", "No") == "No")
```

Comment:
imo, it'd be better to run the check before and after the alert, otherwise it'd be annoying